nix/distribution.nix

```diff
@@ -1,30 +1,44 @@
 {
   inputs,
   self,
-  version,
   ...
 }: {
-  perSystem = {pkgs, ...}: {
-    packages = {
-      inherit (pkgs) prismlauncher-qt5-unwrapped prismlauncher-qt5 prismlauncher-unwrapped prismlauncher;
-      default = pkgs.prismlauncher;
+  perSystem = {
+    lib,
+    pkgs,
+    ...
+  }: {
+    packages = let
+      ourPackages = lib.fix (final: self.overlays.default final pkgs);
+    in {
+      inherit
+        (ourPackages)
+        prismlauncher-qt5-unwrapped
+        prismlauncher-qt5
+        prismlauncher-unwrapped
+        prismlauncher
+        ;
+      default = ourPackages.prismlauncher;
     };
   };
 
   flake = {
     overlays.default = final: prev: let
-      # Helper function to build prism against different versions of Qt.
-      mkPrism = qt:
-        qt.callPackage ./package.nix {
-          inherit (inputs) libnbtplusplus;
-          inherit (prev.darwin.apple_sdk.frameworks) Cocoa;
-          inherit self version;
-        };
+      version = builtins.substring 0 8 self.lastModifiedDate or "dirty";
+
+      # common args for prismlauncher evaluations
+      unwrappedArgs = {
+        inherit (inputs) libnbtplusplus;
+        inherit (prev.darwin.apple_sdk.frameworks) Cocoa;
```

Comment:
@getchoo 
Why did you make this `prev` again?

you only use `prev` for overrides. All other derivation inputs must come from `final`

Reply:
changed it while i was messing around and forgot about how we use the darwin scope :p

this last push is a *little* hacky, but it allows overriding to work as expected with this command (i have no idea why i can't use `recursiveUpdate` here lol)
```
nix eval --json --impure --expr '
  let
  prism = builtins.getFlake "path:/home/seth/repos/PrismLauncher";
  pkgs = import <nixpkgs> {
    system = "x86_64-darwin";
    overlays = [
      prism.overlays.default
      (_: prev: {
        darwin = prev.darwin.overrideScope (_: prev\': {
          apple_sdk = prev\'.apple_sdk // {
            frameworks = prev\'.apple_sdk.frameworks // {Cocoa = prev.emptyFile;};
          };
        });
      })
    ];
  };
  in builtins.elem pkgs.emptyFile pkgs.prismlauncher-unwrapped.buildInputs
  '
  ```